<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/edoardo/catkin_ws/src/modular/urdf/ModularBot_test.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ModularBot" xmlns:xacro="http://ros.org/wiki/xacro">
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <link name="base_link">
    <visual>
      <origin rpy = "0 0 0" xyz = "0 0 0"/>
    </visual>
  </link>
  <link name="L_0">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <box size="0.4 0.4 0.4"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.4 0.4 0.4"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="fixed" type="fixed">
    <parent link="base_link"/>
    <child link="L_0"/>
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
  <link name="L_0_A">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <geometry>
        <cylinder length="0.1" radius="0.06"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.06"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <link name="L_0_B">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <geometry>
        <cylinder length="0.1" radius="0.06"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.1" radius="0.06"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="fixed_A_branch" type="fixed">
    <parent link="L_0"/>
    <child link="L_0_A"/>
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <origin rpy="-1.57 0 0" xyz="0 0.2 0.3"/>
  </joint>
  <joint name="fixed_B_branch" type="fixed">
    <parent link="L_0"/>
    <child link="L_0_B"/>
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <origin rpy="1.57 0 0" xyz="0 -0.2 0.3"/>
  </joint>
</robot>
